docs/standard/serialization/binaryformatter-migration-guide/winforms-migration-guide.md

```diff
@@ -0,0 +1,84 @@
+
+WinForms itself has and still currently contains `BinaryFormatter` code internally for scenarios such as clipboard, drag/drop, and Designer ResX, however, measures have been taken to mitigate the usage of `BinaryFormatter` behind the scenes for common primitive types, including primitive arrays and primitive lists. Usage of these types in clipboard and drag/drop scenarios will continue to work without using `BinaryFormatter` and without any gesture from the developer. Primitive types include the following
+
+- `boolean`
+- `byte`
+- `char`
+- `decimal`
+- `double`
+- `int`
+- `sbyte`
+- `single`
+- `TimeSpan`
+- `DateTime`
+- `uint`
+- `string`
+- `nint`
+- `nuint`
+
+Additional types WinForms currently supports include the following
+
+- `PointF`
+- `RectangleF`
+- `Bitmap`
+- `ImageListStreamer`
+
+## Clipboard
+
+If your clipboard scenario does not involve the types above, `BinaryFormatter` would be used when [`Clipboard.SetData`](https://learn.microsoft.com/dotnet/api/system.windows.clipboard.setdata) is called with your type and when [`Clipboard.GetData`](https://learn.microsoft.com/dotnet/api/system.windows.clipboard.getdata) is called to get your type. `BinaryFormatter` will also be used if [`Clipboard.SetDataObject(object, copy: true)`](https://learn.microsoft.com/dotnet/api/system.windows.clipboard.setdataobject#system-windows-clipboard-setdataobject(system-object-system-boolean)) is called. With the `BinaryFormatter` removal, developers will now see a string about `BinaryFormatter` being removed on the clipboard when using those APIs with types that are not intrinsically handled during serialization/deserialization.
+
+## Drag/Drop
+
+If your drag/drop scenario involves types that are not intrinsically handled during serialization/deserialization, `BinaryFormatter` would be used when [`Control.DoDragDrop`](https://learn.microsoft.com/dotnet/api/system.windows.forms.control.dodragdrop) is called, and the item is dragged out of process and when [`DataObject.GetData`](https://learn.microsoft.com/dotnet/api/system.windows.dataobject.getdata) is called to retrieve your type that has been dragged out of process. With the `BinaryFormatter` removal, developers will now see a string about `BinaryFormatter` being removed upon dropping the dragged item in another process when a drag operation starts with types that are not intrinsically handled.
+
+## The WinForms Designer
+
+The WinForms Designer also uses `BinaryFormatter` internally for ResX serialization/deserialization and CodeDom.
+
+Types and properties may be participating in serialization without developers realizing due to the standard behavior of the WinForms Designer. One way that `BinaryFormatter` is used where developers may not be aware of is when a property on a UserControl is introduced and that control is already populated at design-time, then it’s very likely that that data gets serialized into code or resource files. If all the following are true:
```

Comment:
When are we binary serializing into  code? Sure, custom DesignerSerializer could do whatever they want, but I don't think we ever do it